packages/protobuf/src/extension-accessor.ts

```diff
@@ -29,15 +29,19 @@ import type { Extension } from "./extension.js";
  * default value (if one was specified in the protobuf source), or the zero value
  * (for example `0` for numeric types, `[]` for repeated extension fields, and
  * an empty message instance for message fields).
+ *
+ * Extensions are stored as unknown fields on a message. To mutate an extension
+ * value, make sure to store the new value with setExtension() after mutating.
+ *
+ * If the extension does not extend the given message, an error is raised.
  */
 export function getExtension<E extends Message<E>, V>(
   message: E,
   extension: Extension<E, V>,
   options?: Partial<BinaryReadOptions>,
 ): V {
+  assertExtendee(extension, message);
   const opt = extension.runtime.bin.makeReadOptions(options);
-  assert(!extension.field.oneof); // oneof is not allowed for extensions
-  assert(extension.field.kind != "map"); // maps are not allowed for extensions
```

Comment:
Were these two checks not worth keeping anywhere?

Reply:
These were left over from development, I forgot to remove them earlier. I expect it to very rare that users create their own extension, so we can rely on documentation and typings (makeExtension() does not allow oneof or map) here to shave off non-essential bytes from the bundle size. The assertions for the matching extendee in the accessors is a different situation, because it's very possible to pass in the wrong message without nominal typing.